tests/e2e/e2e_test.go

```diff
@@ -144,14 +144,26 @@ func TestContextMetrics(t *testing.T) {
 	s.Start()
 	defer s.Stop()
 
+	started := false
+	for i := 0; i < 30; i++ {
```

Comment:
Let me play the Gopher pursist for once :P

`NewMetricsServer.Start` could return a `chan bool` and send `true` when ready, then test can wait on `<- start`

LGTM otherwise

Reply:
yes, but then we need "c" to run commands in Start... (Start could also just wait before returning)